From 8423b353cb01015944de895cfef1ef06234206a7 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:40 -0400
Subject: [ethernet] i40e/i40evf: Use dma_set_mask_and_coherent

Message-id: <1409658042-22689-40-git-send-email-sassmann@redhat.com>
Patchwork-id: 90235
O-Subject: [RHEL7.1 PATCH 039/281] i40e/i40evf: Use dma_set_mask_and_coherent
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

In Linux 3.13, dma_set_mask_and_coherent was introduced, and we have
been encouraged to use it. It simplifies the DMA mapping code a bit as
well.

Change-ID: I66e340245af7d0dedfa8b40fec1f5e352754432e
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Catherine Sullivan  <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 6494294f277fdef1409b844b3d6eb1439c3fad8c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 54e1462..7379e5a 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7970,16 +7970,12 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
   return err;
 
  /* set up for high or low dma */
- if (!dma_set_mask(&pdev->dev, DMA_BIT_MASK(64))) {
-  /* coherent mask for the same size will always succeed if
-   * dma_set_mask does
-   */
-  dma_set_coherent_mask(&pdev->dev, DMA_BIT_MASK(64));
- } else if (!dma_set_mask(&pdev->dev, DMA_BIT_MASK(32))) {
-  dma_set_coherent_mask(&pdev->dev, DMA_BIT_MASK(32));
- } else {
-  dev_err(&pdev->dev, "DMA configuration failed: %d\n", err);
-  err = -EIO;
+ err = dma_set_mask_and_coherent(&pdev->dev, DMA_BIT_MASK(64));
+ if (err)
+  err = dma_set_mask_and_coherent(&pdev->dev, DMA_BIT_MASK(32));
+ if (err) {
+  dev_err(&pdev->dev,
+   "DMA configuration failed: 0x%x\n", err);
   goto err_dma;
  }
 
-- 
1.7.1